pojo-generation-core/src/main/java/uk/gov/justice/generation/pojo/plugin/classmodifying/builder/BuilderFieldFactory.java

```diff
@@ -31,6 +34,12 @@
         return fieldDefinitions.stream()
                 .map(fieldDefinition -> {
                     final TypeName typeName = classNameFactory.createTypeNameFrom(fieldDefinition, pluginContext);
+
+                    if (typeName instanceof ParameterizedTypeName && Objects.equals(((ParameterizedTypeName) typeName).rawType.simpleName(), Optional.class.getSimpleName())) {
+                        final TypeName parameterizedType = ((ParameterizedTypeName) typeName).typeArguments.get(0);
+                        return FieldSpec.builder(parameterizedType, fieldDefinition.getFieldName(), PRIVATE).build();
+                    }
+
```

Comment:
This is a bit complex. It may be better to pull this out into its own method/class